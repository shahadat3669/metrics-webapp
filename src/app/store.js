import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {},
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});
