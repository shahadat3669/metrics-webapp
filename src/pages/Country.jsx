import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getCountry } from '../features/countries/countriesSlice';
import styles from '../styles/pages/Country.module.scss';

const Country = () => {
  const { id } = useParams();
  const country = useSelector(getCountry(id));

  return (
    <div className={styles.country}>
      <div className={styles.country__wrapper}>
        <div className={styles.country__top}>
          <div className={styles['country__flag-wrapper']}>
            <img
              className={styles.country__flag}
              src={country?.flag}
              alt={country?.name}
            />
          </div>
          <div className={styles.country__data}>
            <h3 className={styles.country__title}>{country?.name}</h3>
            <p className={styles.country__population}>{country?.population}</p>
          </div>
        </div>
        <div className={styles.country__details}>
          <h4 className={styles['country__details-title']}>Country Details</h4>
          <ul className={styles['country__details-list']}>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Latitude, Longitude
              </div>
              <div className={styles['country__details-item-content']}>
                {`(${country.latlng[0]}, ${country?.latlng[1]})`}
              </div>
            </li>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Total Area
              </div>
              <div className={styles['country__details-item-content']}>
                {country?.area}
              </div>
            </li>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Population
              </div>
              <div className={styles['country__details-item-content']}>
                {country?.population}
              </div>
            </li>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Capital
              </div>
              <div className={styles['country__details-item-content']}>
                {country?.capital[0]}
              </div>
            </li>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Region
              </div>
              <div className={styles['country__details-item-content']}>
                {country.region}
              </div>
            </li>
            <li className={styles['country__details-item']}>
              <div className={styles['country__details-item-title']}>
                Time Zone
              </div>
              <div className={styles['country__details-item-content']}>
                {country?.timezones[0]}
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default Country;
