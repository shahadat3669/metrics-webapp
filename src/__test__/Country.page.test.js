import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Country from '../pages/Country';
import store from '../app/store';

describe('Country', () => {
  test('renders Country Page without crashing', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Country />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree.toJSON).toMatchSnapshot();
  });

  test('renders Country Page without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Country />
        </BrowserRouter>
      </Provider>,
    );
    container.querySelector('.mainTitle');
    expect(container.firstChild).toMatchSnapshot();

    container.querySelector('.first_color');
    expect(container.firstChild).toMatchSnapshot();
  });
});
