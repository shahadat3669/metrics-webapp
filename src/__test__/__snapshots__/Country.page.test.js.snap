// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country renders Country Page without crashing 1`] = `undefined`;

exports[`Country renders Country Page without crashing 2`] = `
<div
  class="country"
>
  <div
    class="country__status"
  >
    Loading...
  </div>
</div>
`;

exports[`Country renders Country Page without crashing 3`] = `
<div
  class="country"
>
  <div
    class="country__status"
  >
    Loading...
  </div>
</div>
`;
